File: java/src/main/java/giis/selema/services/impl/ContainerUtil.java
package giis.selema.services.impl;

import giis.portable.util.JavaCs;
import giis.selema.manager.SelemaException;
import giis.selema.portable.selenium.CommandLine;

public class ContainerUtil {

	private ContainerUtil() {
		throw new IllegalStateException("Utility class");
	}

	/**
	 * Runs a docker command (start or stop), fails if does not respond with the created container name.
	 */
	public static void runDocker(String verb, String container) {
		String command = "docker " + verb + " " + container;
		String dockerOut = CommandLine.runCommand(command);
		if (!container.equals(dockerOut.trim()))
			throw new SelemaException(command + " failed. " + dockerOut);
	}

	/**
	 * Runs a docker command (start or stop) and waits until the container log contains the expected strings.
	 */
	public static void waitDocker(String container, String expectedLog1, String expectedLog2, int timeoutSeconds) {
		// poll the docker log until expected string found
		String logOut = "";
		for (int i = 0; i < timeoutSeconds * 10; i++) {
			logOut = CommandLine.runCommand("docker logs --tail 1 " + container);
			if (logOut.contains(expectedLog1) && logOut.contains(expectedLog2))
				return;
			JavaCs.sleep(100);
		}
		throw new SelemaException("Container did not become ready in time, last log message was: " + logOut);
	}

	/**
	 * Returns the container status (exited, running or paused), error message if it does not exist
	 */
	public static String getContainerStatus(String name) {
		return CommandLine.runCommand("docker inspect -f '{{.State.Status}}' " + name).replace("'", "").trim();
	}

}

File: video-controller/app/containerUtil.js
import { CommandLine } from './commandLine.js';

/**
 * Utility class for Docker container operations.
 */
export class ContainerUtil {
  constructor() {
    throw new Error('Utility class');
  }

  /**
   * Runs a docker command (start or stop), throws if output doesn't match container name.
   * @param {string} verb - Docker verb (e.g., 'start', 'stop').
   * @param {string} container - Container name.
   * @returns {Promise<void>}
   */
  static async runDocker(verb, container) {
    const command = `docker ${verb} ${container}`;
    const dockerOut = await CommandLine.runCommand(command);
    if (dockerOut.trim() !== container) {
      throw new Error(`${command} failed. ${dockerOut}`);
    }
  }

  /**
   * Waits until the container log contains both expected strings.
   * @param {string} container - Container name.
   * @param {string} expectedLog1 - First expected log string.
   * @param {string} expectedLog2 - Second expected log string.
   * @param {number} timeoutSeconds - Timeout in seconds.
   * @returns {Promise<void>}
   */
  static async waitDocker(container, expectedLog1, expectedLog2, timeoutSeconds) {
    let logOut = '';
    const maxTries = timeoutSeconds * 10;

    for (let i = 0; i < maxTries; i++) {
      logOut = await CommandLine.runCommand(`docker logs --tail 1 ${container}`);
      if (logOut.includes(expectedLog1) && logOut.includes(expectedLog2)) {
        return;
      }
      await new Promise(resolve => setTimeout(resolve, 100));
    }

    throw new Error(`Container did not become ready in time, last log message was: ${logOut}`);
  }

  /**
   * Gets the container status (exited, running, paused).
   * @param {string} name - Container name.
   * @returns {Promise<string>}
   */
  static async getContainerStatus(name) {
    const output = await CommandLine.runCommand(`docker inspect -f '{{.State.Status}}' ${name}`);
    return output.replace(/'/g, '').trim();
  }
}
